src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+
+    public void BarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            IconBarConfig config = new IconBarConfig(bar);
+            config.Disabled = !config.Disabled;
+            new SegmentedBarConfig(this).handleBarDisabling(bar);
+        }
+    }
+}
+
+/// <summary>
+///   Configuration class for SegmentedBar class
+/// </summary>
+public class SegmentedBarConfig
+{
+    private SegmentedBar target;
+
+    public SegmentedBarConfig(SegmentedBar target)
+    {
+        this.target = target;
+    }
+
+    public float[] Size
```

Comment:
Like these properties are just a bypass into directly the SegmentedBar. If you want to go through this direct access route (which I advice against, instead you should just set things like the current progress value which internally would set the size then).